R/merge.R

```diff
@@ -9,6 +12,10 @@
         stop("'f' must have the same number of ",dim.type," as 'x'",
             call. = FALSE)
     }
+    # This is done otherwise we lose NA values
+    if ( na.rm ) {
```

Comment:
This should be !na.rm --> By making NA values to characters, we are preserving the observations. Otherwise they are lost. --> NA values are not removed --> !na.rm

Reply:
Also, f can be a factor vector, in that case this will not work. Try something like this

if( !na.rm && any(is.na(f)) ){
    f <- as.character(f)
    f[is.na(f)] <- "NA"
}

Reply:
ACK